src/bosh-director/lib/bosh/director/api/controllers/packages_controller.rb

```diff
@@ -3,81 +3,101 @@
 module Bosh::Director
   module Api::Controllers
     class PackagesController < BaseController
-      post '/matches', :consumes => :yaml do
-        manifest = YAML.load(request.body.read, aliases: true)
+      post '/matches', consumes: :yaml do
+        manifest_hash = YAML.load(request.body.read, aliases: true)
 
-        unless manifest.is_a?(Hash) && manifest['packages'].is_a?(Array)
+        unless manifest_hash.is_a?(Hash) && manifest_hash['packages'].is_a?(Array)
           raise BadManifest, "Manifest doesn't have a usable packages section"
         end
 
-        fingerprint_list = []
+        fingerprints =
+          if existing_release_version_dirty?(manifest_hash)
+            []
+          else
+            fingerprint_list = manifest_hash['packages'].map { |package| package['fingerprint'] }.compact
+            Models::Package.where(fingerprint: fingerprint_list)
+              .where(Sequel.~(sha1: nil))
```

Comment:
could these result in issues where ppl are using the sha256 hack for releases.
where they provide `sha1: sha256:HASH256` ?

Reply:
Possibly - this code no [moved, and not new](https://github.com/cloudfoundry/bosh/pull/2454/files#diff-cec27a0c53f8e3d25c796492e829047ac038ce2f52a56b126d081e64146679f3L23) so if this is an issue we should probably address it in a separate change.

Reply:
The column in the database is always called `sha1`, regardless of whether an actual sha1 or a sha256 is provided. `Sequel.~(sha1: nil)` should translate to SQL `sha1 IS NOT NULL`, so it should cover both sha1 and sha256 cases.

Regardless, this query is not something we changed, simply reordered things to make the structure of both the `/matches` and `/matches_compiled` controller actions similar. The diff is messy, but you can see the query in the previously existing code: https://github.com/cloudfoundry/bosh/blob/5a882fddf3cf0e04ee7be0e74d1fb503b4c59b16/src/bosh-director/lib/bosh/director/api/controllers/packages_controller.rb#L22-L24